File: android-sdk-ui/src/main/java/com/braze/ui/contentcards/view/TextAnnouncementContentCardView.java
package com.braze.ui.contentcards.view;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.appboy.models.cards.TextAnnouncementCard;
import com.appboy.ui.R;
import com.braze.support.StringUtils;

public class TextAnnouncementContentCardView extends BaseContentCardView<TextAnnouncementCard> {
  public TextAnnouncementContentCardView(Context context) {
    super(context);
  }

  private class ViewHolder extends ContentCardViewHolder {
    private final TextView mTitle;
    private final TextView mDescription;

    ViewHolder(View view) {
      super(view, isUnreadIndicatorEnabled());

      mTitle = view.findViewById(R.id.com_braze_content_cards_text_announcement_card_title);
      mDescription = view.findViewById(R.id.com_braze_content_cards_text_announcement_card_description);
    }

    TextView getTitle() {
      return mTitle;
    }

    TextView getDescription() {
      return mDescription;
    }
  }

  @Override
  public ContentCardViewHolder createViewHolder(ViewGroup viewGroup) {
    View view = LayoutInflater.from(viewGroup.getContext())
        .inflate(R.layout.com_braze_text_announcement_content_card, viewGroup, false);
    setViewBackground(view);
    return new ViewHolder(view);
  }

  @Override
  public void bindViewHolder(ContentCardViewHolder viewHolder, final TextAnnouncementCard card) {
    super.bindViewHolder(viewHolder, card);
    ViewHolder textAnnouncementViewHolder = (ViewHolder) viewHolder;

    setOptionalTextView(textAnnouncementViewHolder.getTitle(), card.getTitle());
    setOptionalTextView(textAnnouncementViewHolder.getDescription(), card.getDescription());
    textAnnouncementViewHolder.setActionHintText(StringUtils.isNullOrBlank(card.getDomain()) ? card.getUrl() : card.getDomain());
    viewHolder.itemView.setContentDescription(card.getTitle() + " . " + card.getDescription());
  }
}

File: android-sdk-ui/src/main/java/com/braze/ui/contentcards/view/TextAnnouncementContentCardView.kt
package com.braze.ui.contentcards.view

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.braze.models.cards.Card
import com.braze.models.cards.TextAnnouncementCard
import com.braze.ui.R

open class TextAnnouncementContentCardView(context: Context) :
    BaseContentCardView<TextAnnouncementCard>(
        context
    ) {
    private inner class ViewHolder constructor(view: View) :
        ContentCardViewHolder(view, isUnreadIndicatorEnabled) {
        val title: TextView? = view.findViewById(R.id.com_braze_content_cards_text_announcement_card_title)
        val description: TextView? = view.findViewById(R.id.com_braze_content_cards_text_announcement_card_description)
    }

    override fun createViewHolder(viewGroup: ViewGroup): ContentCardViewHolder {
        val view = LayoutInflater.from(viewGroup.context)
            .inflate(R.layout.com_braze_text_announcement_content_card, viewGroup, false)
        setViewBackground(view)
        return ViewHolder(view)
    }

    override fun bindViewHolder(viewHolder: ContentCardViewHolder, card: Card) {
        if (card is TextAnnouncementCard) {
            super.bindViewHolder(viewHolder, card)
            val textAnnouncementViewHolder = viewHolder as ViewHolder
            textAnnouncementViewHolder.title?.let { setOptionalTextView(it, card.title) }
            textAnnouncementViewHolder.description?.let {
                setOptionalTextView(
                    it,
                    card.description
                )
            }
            val actionHintText = if (card.domain.isNullOrBlank()) card.url else card.domain
            actionHintText?.let { textAnnouncementViewHolder.setActionHintText(it) }
            viewHolder.itemView.contentDescription = "${card.title} . ${card.description}"
        }
    }
}
